File: app/src/main/java/com/example/aac/data/fragment/list/defaults/TestFragmentPresenter.java
package com.example.aac.data.fragment.list.defaults;

import com.aac.expansion.list.AacListFragmentPresenter;

/**
 * Created by yangc on 2017/8/15.
 * E-Mail:yangchaojiang@outlook.com
 * Deprecated:
 */

public class TestFragmentPresenter extends AacListFragmentPresenter<TestListFragment, String> {

    //private TestDataViewModel viewModel;
    @Override
    protected void onCreateView() {
        super.onCreateView();
      //  viewModel = getViewModel(TestDataViewModel.class);
    }

    /**
     * 是否可以加载数据
     * 可以加载数据的条件：
     */
    @Override
    protected void lazyLoad() {
      //  viewModel.getListData(1).observe(getView(), getDataSubscriber());
    }

    @Override
    public void setLoadData(int pager) {
      //  viewModel.getListData(pager).observe(getView(), getDataSubscriber());
    }
}

File: kotlintest/src/main/java/com/example/kotlin/fragment/defaults/TestFragmentPresenter.kt
package com.example.kotlin.fragment.defaults

import com.aac.expansion.list.AacListFragmentPresenter
import com.example.kotlin.viewmodel.TestDataViewModel

/**
 * Created by yangc on 2017/8/15.
 * E-Mail:yangchaojiang@outlook.com
 * Deprecated:
 */

class TestFragmentPresenter : AacListFragmentPresenter<TestListFragment, String>() {
    private var viewModel: TestDataViewModel? = null

    override fun onCreateView() {
        super.onCreateView()
        viewModel = getViewModel(TestDataViewModel::class.java)
    }

    override fun lazyLoad() {
        viewModel!!.getListData(1).observe(view, dataSubscriber)
    }

    override fun setLoadData(pager: Int) {
        viewModel!!.getListData(pager).observe(view, dataSubscriber)
    }

    companion object {
        val TAG = "TestFragmentPresenter"
    }
}
